Add lambda support (#14)
opensearch-project/reporting-cli: docs/examples/lambda_container_image.md

# Scheduling Reports with Lambda

Opensearch reporting CLI can be used with Lambda for scheduling email reports.

### Creating Lambda container image

> 1. Create a Dockerfile. Followig Dockerfile can be used to create a container image for v1.0.0.

Reviewer: ```suggestion
1. Create a Dockerfile. Following Dockerfile can be used to create a container image for v1.0.0.
```

Author: Thanks. fixed typo.

>   ```

Reviewer: ```suggestion
  ```dockerfile
```

  WORKDIR ${FUNCTION_DIR}
  # Copy in the build image dependencies
  COPY --from=build-image ${FUNCTION_DIR} ${FUNCTION_DIR}

  # Install latest chrome dev package and fonts to support major charsets (Chinese, Japanese, Arabic, Hebrew, Thai and a few others)
  # Note: this installs the necessary libs to make the bundled version of Chromium that Puppeteer installs, work.
  RUN apt-get update \
      && apt-get install -y wget gnupg \
      && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
      && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
      && apt-get update \
      && apt-get install -y google-chrome-stable fonts-ipafont-gothic fonts-wqy-zenhei fonts-thai-tlwg fonts-kacst fonts-freefont-ttf libxss1 \
        --no-install-recommends \
      && apt-get remove -y google-chrome-stable \
      && rm -rf /var/lib/apt/lists/*

  ENTRYPOINT ["/usr/local/bin/npx", "aws-lambda-ric"]

  ENV HOME="/tmp"
  CMD [ "/function/src/index.handler" ]
  ```

2. Run the build command in from the directory where Dockerfile exists.

  ```

>   docker build -t opensearch-reporting-cli .

Reviewer: i think users can't run this on arm64 machine, maybe add some requirements on top of the file?

Author: added Prerequisites at the top

  ```
### Use Lambda container image

1. Push the Docker image to [Amazon ECR](https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-console.html)

> 2. Create a Lambda Function. Select previously created container image ro deploy for you function and architecture x86_64'.

Reviewer: probably typo? didn't understand `ro deploy`

Author: yes it was typo. updated this line.

> 3. Update Lambda timeout to 5 min and memory size to 4096.

Reviewer: just curious and for cheaper lambda, does it work if set to 1024? if it works maybe
```suggestion
3. Update Lambda timeout to 5 min and memory size to at least 1024MB.
```

also question how long does it usually take to run in lambda?

Author: I was having issues with chromium on lambda so all articles I read suggested at least 4096, so I kept the same but just tried 1024 and that worked fine.
